testsuite/cluster-test/src/main.rs

```diff
@@ -561,11 +561,14 @@ impl ClusterTestRunner {
         let suite_started = Instant::now();
         for experiment in suite.experiments {
             let experiment_name = format!("{}", experiment);
-            self.run_single_experiment(experiment, None)
+            let experiment_result = self
+                .run_single_experiment(experiment, None)
                 .await
-                .map_err(move |e| {
-                    format_err!("Experiment `{}` failed: `{}`", experiment_name, e)
-                })?;
+                .map_err(move |e| format_err!("Experiment `{}` failed: `{}`", experiment_name, e));
+            if experiment_result.is_err() {
+                self.print_report();
```

Comment:
We could call `self.report.report_text(error details)` here and then when we see that cti failed, try to extract this text, and include it into bot message

Reply:
Good idea. I've updated it to just get the details from the above format_err

Reply:
I am trying to read the code of `land-blocking.yml` and trying to understand whether we are failing correctly in this case? What is the message if this fails?
Any chance we can test this PR to make sure we get proper error message on failure _and_ that bors receives proper failure signal?

I am not sure if there is super simple way to test this, but I think something like this can do the trick:

- put something like `bai!("my error")` in perf benchmark experiment
- try to land this pr with r+
- wait until bors message smth like 'testing commit xxx' and starts gh workflow
- write `r-` to make sure bors don't land this PR even if test pass
- confirm GH workflow actually failed and error message on the PR matches expectations

Reply:
I tried to canary the workflow on-PR to master, but the Github context is not correct, since we rely on the head_commit message, which is only available on-push. https://github.com/libra/libra/pull/5108/checks?check_run_id=872101325. Maybe @sausagee can assist -- how would one test LBT? Would it necessarily be through bors then cancel?

Reply:
I was able to canary using push to other branch and editing the workflow triggers accordingly. https://github.com/libra/libra/runs/874526546?check_suite_focus=true. The job fails as expected. I've gone ahead and added a check for `TypeError` if we try to access a metric that is not available in the report. What do you think @andll ?

```
Cluster Test failed - test failed before pushing metrics.  See https://github.com/libra/libra/actions/runs/170350538
Repro cmd:

    ./scripts/cti --tag land_3245d9b3 --run bench
```

Reply:
Do you think actually adding error message that cluster test reported would be more useful for user?

>Experiment `all up` failed: `Failed to run experiment: TEST INDUCE FAILURE!!!`
See https://github.com/libra/libra/actions/runs/170350538
Repro cmd:
>
>    ./scripts/cti --tag land_3245d9b3 --run bench

Reply:
E.g. the reason to actually include error message in report is that it can later be parsed and included into PR comment

Reply:
I see. Yea that sounds useful. Experiments would have to be explicit about their errors then since otherwise it would be confusing. e.g. in compatibility_test, something like:

```
        info!("1. Changing the images for the first instance to validate storage");
        let first_node = vec![self.first_node.clone()];
        update_batch_instance(context, &first_node, self.updated_image_tag.clone()).await?;
        context
            .tx_emitter
            .emit_txn_for(
                job_duration,
                EmitJobRequest::for_instances(first_node, context.global_emit_job_request),
            )
            .await
            .map_err(|e| anyhow::format_err!("Storage backwards compat broken: {}", e))?;
```

Reply:
Agreed, we'll need better error messages in experiments too

Reply:
Cluster Test Result
```
Experiment `all up` failed: `Failed to run experiment: TEST INDUCE FAILURE!!!`
```

 Cluster Test failed - test report processing failed. See https://github.com/libra/libra/actions/runs/170439368
Repro cmd:

    ./scripts/cti --tag land_40381b6c --run bench

Reply:
Canaried here: https://github.com/libra/libra/runs/874835211?check_suite_focus=true

Reply:
Did you update this PR with code that does this reporting? It seem to be outdated

Reply:
done